src/Nethermind/Nethermind.JsonRpc/Modules/Subscribe/Subscription.cs

```diff
@@ -16,64 +14,27 @@ protected Subscription(IJsonRpcDuplexClient jsonRpcDuplexClient)
         {
             Id = string.Concat("0x", Guid.NewGuid().ToString("N"));
             JsonRpcDuplexClient = jsonRpcDuplexClient;
-            ProcessMessages();
         }
 
         public string Id { get; }
         public abstract string Type { get; }
         public IJsonRpcDuplexClient JsonRpcDuplexClient { get; }
-        private Channel<Action> SendChannel { get; } = Channel.CreateUnbounded<Action>(new UnboundedChannelOptions() { SingleReader = true });
 
         public virtual void Dispose()
         {
-            SendChannel.Writer.Complete();
         }
 
         protected JsonRpcResult CreateSubscriptionMessage(object result)
         {
             return JsonRpcResult.Single(
-                new JsonRpcSubscriptionResponse()
+                new JsonRpcSubscriptionResponse
                 {
-                    Params = new JsonRpcSubscriptionResult()
+                    Params = new JsonRpcSubscriptionResult
                     {
                         Result = result,
                         Subscription = Id
                     }
                 }, default);
         }
-
-        protected void ScheduleAction(Action action)
-        {
-            SendChannel.Writer.TryWrite(action);
-        }
-
-        protected string GetErrorMsg() => $"{Type} subscription with ID {Id} failed.";
-
-        private void ProcessMessages()
-        {
-            Task.Factory.StartNew(async () =>
-            {
-                while (await SendChannel.Reader.WaitToReadAsync())
-                {
-                    while (SendChannel.Reader.TryRead(out Action action))
-                    {
-                        try
-                        {
-                            action();
-                        }
-                        catch (Exception e)
-                        {
-                            if (_logger.IsDebug) _logger.Debug($"{GetErrorMsg()} With exception {e}");
-                        }
-                    }
-                }
-            }, TaskCreationOptions.LongRunning).ContinueWith(t =>
-            {
-                if (t.IsFaulted)
-                {
-                    if (_logger.IsError) _logger.Error($"{GetErrorMsg()} {nameof(ProcessMessages)} encountered an exception.", t.Exception);
-                }
-            });
-        }
```

Comment:
Instead of creating our own ad-hoc event loop we use `async void` event handlers on each subscription, and let C# figure how to schedule these tasks.